### PR TITLE
Add try/catch to Start and Stop DbaXESession functions

### PR DESCRIPTION
**Pull request by dgmyrs**
## Type of Change

 - [x] Bug fix (non-breaking change, fixes #<enter issue number>)
 - [ ] New feature (non-breaking change, adds functionality)
 - [ ] Breaking change (effects multiple commands or functionality)
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [ ] Adding code coverage to existing functionality
 - [ ] Pester test is included
 - [ ] Nunit test is included
 - [ ] Documentation
 - [ ] Build system
 

### Purpose
 
Add try/catch to Stop/Start XESession

### Approach

Write-Error on catch, and return Get-DbaXESession in either case.

### Commands to test

Included in Tests file

### CODE REVIEW

File: functions/Start-DbaXESession.ps1 [hunk 1]
```diff
@@ -86,11 +86,15 @@ function Start-DbaXESession {
 				$session = $xe.Name
 				if (-Not $xe.isRunning) {
 					Write-Message -Level Verbose -Message "Starting XEvent Session $session on $instance."
-					$xe.Start()
-					Get-DbaXESession -SqlInstance $xe.Parent -Session $session
+					try {
+						$xe.Start()
+					} catch {
+						Write-Error -Message "Could not start XEvent Session $session on $instance"
```
Comment (wsmelton) [inline functions/Start-DbaXESession.ps1]: This should use `Stop-Function` this is part of the message system in the module. Supports not showing the sea of read all the time.

So in this case you would have something like `Stop-Function -Message "Could not start XEvent Session on $instance" -Target $session -ErrorRecord $_`

File: functions/Stop-DbaXESession.ps1 [hunk 2]
```diff
@@ -86,11 +86,15 @@ function Stop-DbaXESession {
 				$session = $xe.Name
 				if ($xe.isRunning) {
 					Write-Message -Level Verbose -Message "Stopping XEvent Session $session on $instance."
-					$xe.Stop()
-					Get-DbaXESession -SqlInstance $xe.Parent -Session $session
+					try {
+						$xe.Stop()
+					} catch {
+						Write-Error -Message "Could not stop XEvent Session $session on $instance"
```
Comment (wsmelton) [inline functions/Stop-DbaXESession.ps1]: Same as previous, need to use `Stop-Function`.

File: functions/Start-DbaXESession.ps1 (hunk 1 above)
Comment (dgmyrs) [reply functions/Start-DbaXESession.ps1]: OK.  I added a commit with the changes.

File: functions/Stop-DbaXESession.ps1 [hunk 3]
```diff
@@ -89,7 +89,7 @@ function Stop-DbaXESession {
 					try {
 						$xe.Stop()
 					} catch {
-						Write-Error -Message "Could not stop XEvent Session $session on $instance"
+						Stop-Function -Message "Could not stop XEvent Session on $instance" -Target $session -ErrorRecord $_
```
Comment (wsmelton) [inline functions/Stop-DbaXESession.ps1]: One more minor adjustment I didn't catch until I looked at the whole command. We need to add `-Continue` switch to `Stop-Function` call so it won't stop the foreach loop and keep going to the next one in the collection. If you can do that on each one I'll get this merged.

Comment (dgmyrs) [reply functions/Stop-DbaXESession.ps1]: ah, yup, makes sense.  Done in the latest commit

**Pull request merged**